sdk/src/types/block/address/bech32.rs

```diff
@@ -216,17 +223,81 @@ impl PartialEq<str> for Bech32Address {
     }
 }
 
+impl PartialEq<Bech32AddressString> for Bech32Address {
+    fn eq(&self, other: &Bech32AddressString) -> bool {
+        self.to_string().eq(&other.0)
```

Comment:
it's probably slightly more future proof if you use `to_bech32_address_string` instead of `to_string`.